crates/sui-rpc-loadgen/src/payload/rpc_command_processor.rs

```diff
@@ -246,41 +256,92 @@ impl<'a> ProcessPayload<'a, &'a GetCheckpoints> for RpcCommandProcessor {
+            let start_time = Instant::now();
+            let transactions = client
+                .read_api()
+                .multi_get_transactions_with_options(
+                    digests.to_vec(),
+                    SuiTransactionResponseOptions::full_content(), // todo(Will) support options for this
+                )
+                .await;
+            let elapsed_time = start_time.elapsed();
+            debug!(
+                "MultiGetTransactions Request latency {:.4} for rpc at url {i}",
+                elapsed_time.as_secs_f64()
+            );
+            transactions
+        },
+    ))
+    .await;
+
+    if record_addresses {
+        // TODO: apply this logic to transactions_response directly, which should simplify the rest of the code and support validating more than 2
+        let transactions = transactions_response
             .iter()
             .enumerate()
-            .map(|(i, client)| async move {
-                let start_time = Instant::now();
-                let transactions = client
-                    .read_api()
-                    .multi_get_transactions_with_options(
-                        digests.to_vec(),
-                        SuiTransactionResponseOptions::full_content(), // todo(Will) support options for this
-                    )
-                    .await;
-                let elapsed_time = start_time.elapsed();
-                debug!(
-                    "MultiGetTransactions Request latency {:.4} for rpc at url {i}",
-                    elapsed_time.as_secs_f64()
-                );
-                transactions
-            }),
-    )
-    .await;
+            .filter_map(|(i, result)| match result {
+                Ok(transactions) => Some(transactions),
+                Err(err) => {
+                    warn!(
+                        "Failed to fetch transactions for vec {i}: {:?}. Logging digests, {:?}",
+                        err, digests
+                    );
+                    None
+                }
+            })
+            .cloned();
+
+        let addresses: Vec<SuiAddress> = transactions
+            .into_iter()
+            .flat_map(|vec_transactions| {
+                vec_transactions.into_iter().filter_map(|transaction_item| {
+                    transaction_item
+                        .transaction
+                        .map(|sui_transaction| match sui_transaction.data {
+                            SuiTransactionData::V1(data_v1) => data_v1.sender,
+                        })
+                })
+            })
+            .collect();
+
+        debug!("Recording addresses {:?}", addresses.len());
+        let address_set: HashSet<SuiAddress> = HashSet::from_iter(addresses);
+        match write_data_to_file(&address_set, &format!("{file_path}/addresses")) {
+            Ok(_) => debug!("Successfully written addresses"),
+            Err(err) => error!("Failed to write addresses: {:?}", err),
+        }
```

Comment:
I think a better way is to use a concurrent data structure like a `DashMap` here, that way we can ensure:
1. all addresses are unique across all transactions processed by all threads
2. more efficient than writing/creating a file for every transaction

It's possible that the dashmap can exceed memory size at some point, but we can tackle this problem later as a separate PR by dumping it into a file when the size becomes too large.